pkg/daemon/daemon.go

```diff
@@ -927,6 +927,7 @@ func (dn *Daemon) checkStateOnFirstRun() error {
 	if state.pendingConfig != nil && bootID == dn.bootID {
 		dn.logSystem("drain interrupted, retrying")
 		if err := dn.drain(); err != nil {
+			dn.recorder.Eventf(getNodeRef(dn.node), corev1.EventTypeWarning, "FailedToDrain", err.Error())
```

Comment:
it'll avoid duplication if you fire that event directly from `dn.drain()` so every time it fails it fires an event

Reply:
and if we add another drain call somewhere else in the future it'll be future proof as well w/o forgetting

Reply:
Good point. Done.